packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -529,15 +533,23 @@ class RangeMaintainingScrollPhysics extends ScrollPhysics {
       enforceBoundary = false;
     }
     if (maintainOverscroll) {
-      // Force the new position to be no more out of range
-      // than it was before, if it was overscrolled.
-      if (oldPosition.pixels < oldPosition.minScrollExtent) {
-        final double oldDelta = oldPosition.minScrollExtent - oldPosition.pixels;
-        return newPosition.minScrollExtent - oldDelta;
-      }
-      if (oldPosition.pixels > oldPosition.maxScrollExtent) {
-        final double oldDelta = oldPosition.pixels - oldPosition.maxScrollExtent;
-        return newPosition.maxScrollExtent + oldDelta;
+      // Force the new position to be no more out of range than it was before, if:
+      //  * it was overscrolled
+      //  * and the position has shrunk, meaning that some content was removed. The
```

Comment:
trivial nit: it's conventional to have `, and` at the end of the first bullet rather than the start of the second.

Reply:
Done